auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java

```diff
@@ -137,6 +143,49 @@ private static boolean needsContentDescriptor(Context context) {
     return true;
   }
 
+  private static boolean needsWriteToParcel(Context context) {
+    ProcessingEnvironment processingEnv = context.processingEnvironment();
+    TypeMirror parcel = processingEnv.getElementUtils().getTypeElement("android.os.Parcel").asType();
+    ProcessingEnvironment env = context.processingEnvironment();
+    for (ExecutableElement element : MoreElements.getLocalAndInheritedMethods(
+            context.autoValueClass(), env.getElementUtils())) {
+      if (element.getSimpleName().contentEquals("writeToParcel")
+              && MoreTypes.isTypeOf(void.class, element.getReturnType())
+              && !element.getModifiers().contains(Modifier.ABSTRACT)) {
+        List<? extends VariableElement> parameters = element.getParameters();
+        if (parameters.size() == 2
+                && processingEnv.getTypeUtils().isSameType(parcel, parameters.get(0).asType())
+                && MoreTypes.isTypeOf(int.class, parameters.get(1).asType())) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  private static boolean needsCreator(Context context) {
+    ProcessingEnvironment processingEnv = context.processingEnvironment();
+    Types typeUtils = processingEnv.getTypeUtils();
+    Elements elementUtils = processingEnv.getElementUtils();
+    TypeMirror creatorType = typeUtils.erasure(elementUtils.getTypeElement("android.os.Parcelable.Creator").asType());
+    List<? extends Element> members = processingEnv.getElementUtils().getAllMembers(context.autoValueClass());
+    for (VariableElement field : ElementFilter.fieldsIn(members)) {
+      if (field.asType() instanceof DeclaredType) {
+        List<? extends TypeMirror> typeArguments = ((DeclaredType) field.asType()).getTypeArguments();
+        if (typeArguments.size() == 1
+                && field.getSimpleName().contentEquals("CREATOR")
+                && typeUtils.isSameType(creatorType, typeUtils.erasure(field.asType()))
+                && typeUtils.isAssignable(context.autoValueClass().asType(), typeArguments.get(0))
+                && field.getModifiers().contains(Modifier.STATIC)
+                && field.getModifiers().contains(Modifier.PUBLIC)
+                && field.getModifiers().contains(Modifier.FINAL)) {
```

Comment:
I'm not 100% if I should be checking for FINAL here. I'm also not 100% on whether or not I should be checking the type argument of the creator either. Both of those things aren't mandatory. Thoughts?